org.epsg.openconfigurator/src/org/epsg/openconfigurator/xmloperation/ProjectJDomOperation.java

```diff
@@ -418,15 +422,42 @@ private static void addRedundantManagingNode(Document document, TRMN node) {
 
     public static void deleteModule(Document document, Module module,
             boolean finalModuleCheck) {
-        System.err.println("finalModuleCheck....." + finalModuleCheck);
-        if (!finalModuleCheck) {
+        Node node = module.getNode();
+        int availableModule = VALUE_ZERO;
+        List<HeadNodeInterface> headNodeInterfaceList = node
+                .getHeadNodeInterface();
+        if (headNodeInterfaceList.size() != VALUE_ZERO) {
+            for (HeadNodeInterface headNodeInterface : headNodeInterfaceList) {
+                Collection<Module> moduleCollection = headNodeInterface
+                        .getModuleCollection().values();
+                if (moduleCollection != null) {
+                    for (Module mod : moduleCollection) {
```

Comment:
Useless code -> moduleCollection.size() tells whether there are modules or not.
Counting in a for loop is

Reply:
In this code snippet, moduleCollection.size() checks the availability of modules for only one interface.

For example in the below case, CN(1) is modular head node with single Interface and multiple modules. moduleCollection.size() checks availability of modules only for the interface IF1

MN(240)
--->CN(1)
------->IF1
----------->M1
----------->M2
----------->M3
----------->M4

If a modular node has multiple interfaces (for example: IF1, IF2 etc..), this counting in for loop enables us to identify total number of modules connected to all the interfaces of modular head node. In the below case, CN(2) is modular head node with multiple Interface.

MN(240)
--->CN(1)
--->CN(2)
------->IF1
----------->M1
------->IF2
----------->M2
------->IF3
----------->M3
------->IF4
----------->M4
------->IF5
----------->M5
------->IF6
----------->M6

The purpose of counting modules in for loop is to remove the <InterfaceList> tag from project XML file, if there are no modules. If this is not handled, it will result in project file schema error.